presto-manager-agent/src/main/java/com/teradata/prestomanager/agent/AgentServerModule.java

```diff
@@ -40,7 +41,24 @@ protected void setup(Binder binder)
     {
         binder.disableCircularProxies();
 
+        // buildConfigObject also calls bindConfig
+        AgentConfig config = buildConfigObject(AgentConfig.class);
+        configBinder(binder).bindConfig(PrestoConfig.class, "presto");
+
+        switch (config.getPackageType()) {
```

Comment:
`Module` is a functional interface, so I don't even need more classes.

The big downside here is that you can't use this to throw an exception when an invalid package type is given, besides manually checking the configuration like we already are. (Though if none of the conditional modules are bound, it will cause a exception when creating the injector when `PackageController` has no explicit bindings.)